lumicks/pylake/channel.py

```diff
@@ -583,13 +584,51 @@ def _apply_mask(self, mask):
         return TimeSeries(self.data[mask], self.timestamps[mask])
 
     @staticmethod
-    def from_dataset(dset, y_label="y", calibration=None):
+    def from_dataset(dset, y_label="y", calibration=None) -> Slice:
+        class LazyLoadedCompoundField:
+            """Wrapper to enable lazy loading of HDF5 compound datasets
+
+            Notes
+            -----
+            We only need to suppot the methods `__array__()` and `__len__()`, as we only access
```

Comment:
I appreciate the little note for our future selves.

Minor typo though: `suppot` -> `support`

Reply:
fixed